packages/mermaid/src/diagrams/git/gitGraphAst.ts

```diff
@@ -84,54 +106,72 @@ function uniqBy(list, fn) {
   }, []);
 }
 
-export const setDirection = function (dir) {
+export const setDirection = function (dir: DiagramOrientation) {
   direction = dir;
 };
+
 let options = {};
-export const setOptions = function (rawOptString) {
+export const setOptions = function (rawOptString: string) {
   log.debug('options str', rawOptString);
   rawOptString = rawOptString && rawOptString.trim();
   rawOptString = rawOptString || '{}';
   try {
     options = JSON.parse(rawOptString);
   } catch (e) {
-    log.error('error while parsing gitGraph options', e.message);
+    if (e instanceof Error) {
+      log.error('error while parsing gitGraph options', e.message);
+    } else {
+      throw e;
+    }
   }
 };
 
 export const getOptions = function () {
   return options;
 };
 
-export const commit = function (msg, id, type, tag) {
+export const commit = function (msg: string, id: string, type: any, tag: string) {
   log.debug('Entering commit:', msg, id, type, tag);
   id = common.sanitizeText(id, getConfig());
   msg = common.sanitizeText(msg, getConfig());
   tag = common.sanitizeText(tag, getConfig());
-  const commit = {
+  const commit: Commit = {
     id: id ? id : seq + '-' + getId(),
     message: msg,
     seq: seq++,
-    type: type ? type : commitType.NORMAL,
+    type: type ? type : CommitType.NORMAL, // TODO
     tag: tag ? tag : '',
     parents: head == null ? [] : [head.id],
-    branch: curBranch,
+    branch: currBranchName,
   };
   head = commit;
   commits[commit.id] = commit;
-  branches[curBranch] = commit.id;
+  branches[currBranchName] = commit.id;
   log.debug('in pushCommit ' + commit.id);
 };
 
-export const branch = function (name, order) {
+export const branch = function (name: string, order: string) {
   name = common.sanitizeText(name, getConfig());
   if (branches[name] === undefined) {
+    const orderInt = parseInt(order, 10);
+    if (isNaN(orderInt)) {
+      throw new Error('Invalid order provided');
+      // TODO - what is this hash?
```

Comment:
The hash is used in external apps like the live editor to show more info about an error.

Reply:
Oh so in your previous comment about removing that hash, you didn't mean that I can remove all the hashes?

Reply:
Bump on this @sidharthv96

Reply:
You can remove the ones that are incorrect/filled with dummy data. It would be better if you can put the right data in, but I don't think that's available in the DB. (line position and all)